core/lib/instrument/src/alloc/mod.rs

```diff
@@ -0,0 +1,37 @@
+//! Allocation instrumentation.
+
+use std::fmt;
+
+use self::guard_impl::AllocationGuardImpl;
+
+#[cfg(feature = "jemalloc")]
+mod guard_impl;
+#[cfg(not(feature = "jemalloc"))]
+#[path = "mock_guard_impl.rs"]
+mod guard_impl;
```

Comment:
This is awkward to maintain, but I don't see an alternative if we plan to support Jemalloc conditionally in node binaries, as it's defined rn.